crates/ruff_python_ast/src/nodes.rs

```diff
@@ -781,6 +831,92 @@ impl From<ExprDict> for Expr {
     }
 }
 
+#[derive(Debug)]
+pub struct DictKeyIterator<'a> {
+    items: Iter<'a, DictItem>,
+}
+
+impl<'a> DictKeyIterator<'a> {
+    fn new(items: &'a [DictItem]) -> Self {
+        Self {
+            items: items.iter(),
+        }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+}
```

Comment:
You get `is_empty` for free by implementing `is_empty`. You may want to consider overriding `ExactSizeIterator::len()`

Reply:
Unfortunately `is_empty` is an unstable feature of `ExactSizeIterator` currently: https://doc.rust-lang.org/std/iter/trait.ExactSizeIterator.html#method.is_empty